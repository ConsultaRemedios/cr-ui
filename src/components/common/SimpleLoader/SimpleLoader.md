A loader that can be used anywhere.

```js
new Vue({
  data(){
    return { isLoading: false };
  },

  methods: {
    toggleLoader() {
      this.isLoading = !this.isLoading;
    }
  },

  template: `
    <div>
      <BaseButton @click="toggleLoader">Toggle SimpleLoader</BaseButton>
      <SimpleLoader v-if="isLoading" />
    </div>
  `
})
```

You can pass a message too.

```js
new Vue({
  data(){
    return { isLoading: false };
  },

  methods: {
    toggleLoader() {
      this.isLoading = !this.isLoading;
    }
  },

  template: `
    <div>
      <BaseButton @click="toggleLoader">Toggle SimpleLoader</BaseButton>
      <SimpleLoader v-if="isLoading">Loading...</SimpleLoader>
    </div>
  `
})
```
