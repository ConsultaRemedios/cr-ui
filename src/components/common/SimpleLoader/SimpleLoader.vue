<template>
  <div :class="$style.content">
    <img :class="$style.image" src="./../../../assets/loader.svg" />

    <span v-if="hasSlotData" :class="$style.text">
      <slot></slot>
    </span>
  </div>
</template>

<script>
  export default {
    name: 'SimpleLoader',

    computed: {
      hasSlotData() {
        return this.$slots.default;
      },
    },
  };
</script>

<style module>
  .image {
    display: block;
    margin: 0 auto;
    animation: rotate-loader 1.5s infinite;
  }

  .text {
    display: block;
    max-width: 200px;
    margin: 10px auto 0;

    font-weight: 600;
    font-size: 16px;
    line-height: 1.3;
    text-align: center;
  }

  @keyframes rotate-loader {
    0% {
      transform: rotate(0deg);
    }

    100% {
      transform: rotate(360deg);
    }
  }
</style>
