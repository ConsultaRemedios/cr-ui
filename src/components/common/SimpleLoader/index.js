import SimpleLoader from './SimpleLoader.vue';

export default SimpleLoader;
