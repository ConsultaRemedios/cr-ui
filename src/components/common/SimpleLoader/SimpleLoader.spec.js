import { shallowMount } from '@vue/test-utils';
import SimpleLoader from './SimpleLoader.vue';

const mountSimpleLoader = (slot = []) => {
  return shallowMount(SimpleLoader, {
    slots: {
      default: slot,
    }
  });
}

describe('SimpleLoader component', () => {
  describe('When its empty', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = mountSimpleLoader();
    });

    it('matches the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('When has some slot', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = mountSimpleLoader('Some text');
    });

    it('matches the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('Computed', () => {
    describe('#hasSlotData', () => {
      describe('When has no slot', () => {
        let wrapper;

        beforeAll(() => {
          wrapper = mountSimpleLoader();
        });

        it('return "false"', () => {
          expect(wrapper.vm.hasSlotData).toBeFalsy();
        });
      });

      describe('When has slot', () => {
        let wrapper;

        beforeAll(() => {
          wrapper = mountSimpleLoader('Some text');
        });

        it('return "true"', () => {
          expect(wrapper.vm.hasSlotData).toBeTruthy();
        });
      });
    });
  });
});
