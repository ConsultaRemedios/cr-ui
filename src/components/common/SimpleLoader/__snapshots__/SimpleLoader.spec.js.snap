// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleLoader component When has some slot matches the snapshot 1`] = `
<div>
  <img
    class="image"
    src="./../../../assets/loader.svg"
  />
   
  <span
    class="text"
  >
    Some text
  </span>
</div>
`;

exports[`SimpleLoader component When its empty matches the snapshot 1`] = `
<div>
  <img
    class="image"
    src="./../../../assets/loader.svg"
  />
   
  <!---->
</div>
`;
